### PR TITLE
PR hygiene checking

### LINKED ISSUE #782: Update index.html title not very helpful
Github bases issue names based on the title. Because all our files are named index.html we end up with lots of issues named   **Update index.html**. 

Can we trick github to grab the name from the parent folder by default?

### PR DESCRIPTION
**Pull request by peterbe**
fixes #782 

Here's what it looks like if the PR title was `Update index.html`
Screen Shot 2021-01-04 at 10 37 24 AM

And if all is well:
Screen Shot 2021-01-04 at 10 37 56 AM

Not rocket science but hopefully a start.

### CODE REVIEW

**Comment by nschonni** [comment]
https://github.com/marketplace/actions/conventional-pr-title-alt might be another interesting option

**Comment by peterbe** [comment]
> https://github.com/marketplace/actions/conventional-pr-title-alt might be another interesting option
> 
> **Edit:** or one of these https://github.com/marketplace?type=actions&query=title

I looked at most of those and didn't feel attracted to any of them. They all seemed overly complicated or just not perfect. And besides, I quite like the idea of building it "from scratch" because then you're limitless in what you want to implement. Our current logic is a bit odd in that it's so oddly specific. And we can now quite easily express our business logic with simple string-checking if-statements in JavaScript rather than custom regexes in a Yaml file.

File: pr-lint/src/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    return (
+      'Pull request title can\'t just be "Update index.html"\n' +
+      "Please update the pull request to be more descriptive. " +
+      "For example 'fix typo on Web/JavaScript'."
+    );
+  }
+
+  const { body } = contextPullRequest;
+  if (!body.trim()) {
+    // Even for typos it's required that to explain something about the PR.
```
Comment (escattone) [inline pr-lint/src/index.js:36]:
```suggestion
    // Even for typos it's required to explain something about the PR.
```

**Review by escattone — APPROVED** [review]
This seems like a good start to me, thanks @peterbe!

**Comment by peterbe** [comment]
ping @fiji-flo

**Comment by peterbe** [comment]
One thing I've found for myself is that when I do a lot of flaw fixing PRs, I never type in a description. Because it's rarely useful when the summary is sufficient.

**Comment by Ryuno-Ki** [comment]
This looks ready to merge (from a CI point of view)

**Comment by peterbe** [comment]
I changed my mind. I made it less controversial so that it doesn't require that the description is non-empty. 
I've made many PRs on the content and I often find that once you've typed a decent title, there's really nothing else worth saying in the description. At least not worthy enough for it to block in CI.

**Comment by peterbe** [comment]
@fiji-flo Do you mind if I merge without your review?

**Pull request merged**